specs/AppNotifications/AppNotificationContentSpec/AppNotificationBuilder-spec.md

```diff
@@ -0,0 +1,668 @@
+```c#
+AppNotificationContent()
+    .AddSelectionMenu(SelectionMenu(L"selectionMenu")
+        .SetDefaultSelection(L"yes")
+        .AddSelection(L"yes", L"Going")
+        .AddSelection(L"maybe", L"Maybe")
+        .AddSelection(L"no", L"Decline"))
+    .AddButton(Button(L"Send")
+        .SetInputId(L"selectionMenu"))
+    .Show();
+```
+
+![SelectionMenu Example](SelectionMenuExample.png)
+
+# API Details
+
+```c#
+namespace Microsoft.Windows.AppNotifications.Builder
+{
+    runtimeclass Text
+    {
+        Text(String content);
+
+        String Content{ get; };
+        Text SetContent(String content);
+
+        // The target locale of the XML payload, specified as a BCP-47 language tags.
+        String Language{ get; };
+        Text SetLanguage(String language);
+
+        // Center the text for incoming call notifications. This value is only used for notifications with with a scenario value of IncomingCall.
+        Boolean IsCallScenario { get; };
+        Text SetIsCallScenario(Boolean isCallScenario);
+
+        // Displays text at the bottom of your notification, along with your app's identity or the notification's timestamp.
+        // On older versions of Windows that don't support attribution text, the text will simply be displayed as another text element.
+        Boolean IsAttributionText { get; };
+        Text SetIsAttributionText(Boolean isAttributionText);
+
+        // Retrieves the XML content of the Text.
+        String GetXml();
+    }
+
+    enum ButtonStyle
+    {
+        Success, // The button will be green
+        Critical, // The button will be red
+    }
+
+    runtimeclass Button
+    {
+        Button();
+        Button(String content);
+
+        // Get/Sets the text to display on the button.
+        String Content{ get; };
+        Button SetContent(String content);
+
+        // Gets app-defined string of arguments that the app can later retrieve once it is activated when the user clicks the button.
+        String Arguments { get; };
+
+        // Gets/sets the Icon source for the button.
+        String IconUri { get; };
+        Button SetIconUri(String iconUri);
+
+        // The tooltip for a button, if the button has an empty content string.
+        String ToolTip { get; };
+        Button SetToolTip(String toolTip);
+
+        // Sets if the Button will be a context menu action.
+        Boolean IsContextMenuPlacement { get; };
+        Button SetIsContextMenuPlacement(Boolean isContextMenuPlacement);
+
+        ButtonStyle ButtonStyle { get;};
+        Button SetButtonStyle(ButtonStyle buttonStyle);
+
+        // Gets or sets the ID of an existing AppNotificationTextBox in order to have this button display to the right of the input.
+        String InputId{ get; };
+        Button SetInputId(String inputId);
+
+        // Adds a key(without value) to the activation arguments that will be returned when the toast notification or its buttons are clicked.
+        Button AddArgument(String key);
```

Comment:
Why is this needed? How is this supposed to be used? Is there an example scenario where a devs would call this?

This applies to all similar method

Reply:
SetInputId is very important because it basically maps the button to a TextBox || SelectionBox for quick reply scenarios.